File: 01/ex2/main.py

numbers = [int(num.strip()) for num in open('input.txt', 'r')]

product = lambda a, b, c : a*b*c if (a+b+c == 2020) else 0

for a in numbers:
    for b in numbers:
        for c in numbers:
            if (tot := product(a, b, c)) > 0:
                print(tot)

File: 01/ex2/main.hs
import System.IO.Unsafe

-- Last step: find the correct numbers
findSum :: [Int] -> Int
findSum xs = head [a*b*c | a <- xs, b <- xs, c <- xs, a + b + c == 2020]


-- PARSE AND READ THE FILE
textFromFile :: String -> String
textFromFile file = do unsafePerformIO (readFile file)

convertToInts :: String -> [Int]
convertToInts s = [read num :: Int | num <- init (lines s)]


-- Main function, tie everything together
answer :: Int
answer = findSum (convertToInts (textFromFile "input.txt"))
